lua/entities/gmod_wire_expression2/core/find.lua

```diff
@@ -442,6 +442,29 @@ e2function entity findPlayerByName(string name)
 	return findPlayer(name)
 end
 
+--- Returns the player with the given SteamID
+e2function entity findPlayerBySteamID(string id)
+	if query_blocked(self, 1) then return NULL end
+	return player.GetBySteamID(id) or NULL
+end
+
+--- Returns the player with the given SteamID64
+e2function entity findPlayerBySteamID64(string id)
+	if query_blocked(self, 1) then return NULL end
+	return player.GetBySteamID64(id) or NULL
+end
+
+--- Returns the player with the given UserID
+e2function entity findPlayerByUserID(number id)
+	if query_blocked(self, 1) then return NULL end
+	for k, v in ipairs(player.GetAll()) do
```

Comment:
Using something such as https://gist.github.com/bigdogmat/9b0cffff16f568ed44e7660520b6dea7, I'm able to get these results

```
Local next:     0.01167373054102
Global next:    0.0062933406899219
Local pairs:    0.011479684652045
Global pairs:   0.0052506362746954
Local ipairs:   0.00049300514251627
Global ipairs:  0.00047430192432785
Numeric for/length operator:    0.00028347065097023
Numeric for:    0.00027733365755012
```

Guessing the reason globals are preforming faster than locals is because of how the jit compiler works, scope indexing isn't much you have to worry about when it comes to working with LuaJIT. I'm sure over time they could become faster due to code normalization.